XamarinCommunityToolkit/Converters/StateToBooleanConverter.shared.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.Forms;
+using Xamarin.CommunityToolkit.UI.Views;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	public class StateToBooleanConverter : IValueConverter
+	{
+		public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+			=> value is State state && parameter is State stateToCompare && state == stateToCompare;
```

Comment:
1) if value is not State, then raise exception
2) try to get State from parameter (if no luck then go ahead with StateToCompare property)

I believe we implemented a similar approach :0